src/renderer/js/component/shapeShift/internal/active-shift.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { PureComponent } from "react";
+import QRCode from "qrcode.react";
+import * as statuses from "constants/shape_shift";
+import Address from "component/address";
+import Link from "component/link";
+
+export default class ActiveShapeShift extends PureComponent {
+  constructor() {
+    super();
+    this.continousFetch = undefined;
+  }
+
+  componentDidMount() {
+    const { getActiveShift, shiftDepositAddress } = this.props;
+
+    getActiveShift(shiftDepositAddress);
+    this.continousFetch = setInterval(() => {
+      getActiveShift(shiftDepositAddress);
+    }, 10000);
+  }
+
+  componentWillUpdate(nextProps) {
+    const { shiftState } = nextProps;
+    if (shiftState === statuses.COMPLETE) {
+      this.clearContinuousFetch();
+    }
+  }
+
+  componentWillUnmount() {
+    this.clearContinuousFetch();
+  }
+
+  clearContinuousFetch() {
+    clearInterval(this.continousFetch);
+    this.continousFetch = null;
+  }
+
+  render() {
+    const {
+      shiftCoinType,
+      shiftDepositAddress,
+      shiftReturnAddress,
+      shiftOrderId,
+      shiftState,
+      shiftDepositLimit,
+      clearShapeShift,
+      doShowSnackBar,
+    } = this.props;
+
+    return (
+      <div>
+        {shiftState === statuses.NO_DEPOSITS && (
+          <div>
+            <p>
+              Send up to{" "}
+              <span className="credit-amount--bold">
+                {shiftDepositLimit}{" "}
+                <span className="shapeshift__success">{shiftCoinType}</span>
+              </span>{" "}
+              to the address below
+            </p>
+
+            <div className="shapeshift__deposit-address-wrapper">
+              <Address address={shiftDepositAddress} showCopyButton />
+              <div className="shapeshift__qrcode">
+                <QRCode value={shiftDepositAddress} />
+              </div>
+            </div>
+          </div>
+        )}
+
+        {shiftState === statuses.RECEIVED && (
+          <div className="shapeshift__received">
+            <p>
+              ShapeShift has received your payment! Sending the funds to your
+              LBRY wallet.<br />
+              <span className="help">
+                {__("This can take a while, especially with BTC")}
+              </span>
+            </p>
+          </div>
+        )}
+
+        {shiftState === statuses.COMPLETE && (
+          <div className="shapeshift__complete">
+            <p>
+              {__(
+                "Transaction complete! You should see the new LBC in your wallet."
+              )}
+            </p>
+          </div>
+        )}
+        <div className="shapeshift__actions">
+          <Link
+            button="alt"
+            onClick={clearShapeShift}
+            label={
+              shiftState === statuses.COMPLETE ||
+              shiftState === statuses.RECEIVED
+                ? __("New")
```

Comment:
I'm not in love with "New" here - maybe just "Reset"? Though I don't like this either. As a (potentially weird) user, I have a desire for things to be back in their default state when "Done". But I don't necessary want to do a new one. I just want it back to where it was. Not sure on best name for this.

Reply:
Also this should probably be primary, whatever the label is.